src/state/chainstate.rs

```diff
@@ -100,12 +128,39 @@ impl ChainstateManager {
         Ok(Self { inner })
     }
 
-    /// Process and validate the passed in block with the [`ChainstateManager`]
-    /// If processing failed, some information can be retrieved through the status
-    /// enumeration. More detailed validation information in case of a failure can
-    /// also be retrieved through a registered validation interface. If the block
-    /// fails to validate the `block_checked` callback's ['BlockValidationState'] will
-    /// contain details.
+    /// Process and validate a block.
+    ///
+    /// Attempts to validate and add the block to the chain. The block goes
+    /// through full consensus validation including proof-of-work, transaction
+    /// validity, and script verification.
+    ///
+    /// # Arguments
+    /// * `block` - The [`Block`] to process
+    ///
+    /// # Returns
+    /// A [`ProcessBlockResult`] indicating whether the block was:
+    /// - Newly accepted and added to the chain ([`ProcessBlockResult::NewBlock`])
```

Comment:
Confusingly, new block does not indicate whether the block was added to the chain. It is only used to indicate if it is committed to disk.

I think it would be good to mention that to good the full validity information, users are required to register with the validation interface's block checked callback.